math/ipopt/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+############################################################################
+# CMakeLists.txt file for building ROOT math/ipopt package
+############################################################################
+
+include_directories(${IPOPT_INCLUDE_DIRS} ${CMAKE_SOURCE_DIR})
+add_definitions(-DUSE_ROOT_ERROR ${IPOPT_DEFINITIONS} )
+
+
+set(headers Math/IpoptMinimizer.h )
+set(linkdefs Math/LinkDef.h )
+
+ROOT_STANDARD_LIBRARY_PACKAGE(Ipopt
+                              HEADERS ${headers}
+                              LINKDEF ${linkdefs}
+                              LIBRARIES ${IPOPT_LIBRARIES}
+                              DEPENDENCIES MathCore )
```

Comment:
Please try to use our new style for build system packages as shown in [sql/mysql/CMakeLists.txt](https://github.com/root-project/root/blob/master/sql/mysql/CMakeLists.txt) or [net/davix/CMakeLists.txt](https://github.com/root-project/root/blob/master/net/davix/CMakeLists.txt). That is, do not use file globbing and global commands like `include_directories()`, `add_definitions()`, etc, but their `target_...` equivalent.